src/formatDate.js

```diff
@@ -50,7 +50,98 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitedDate = date.split(fromFormat[3]);
+  let year = '';
+  let month = '';
+  let day = '';
+
+  for (let i = 0; i < toFormat.length; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+        year = splitedDate[i];
+        break;
+
+      case 'YY':
+        year = splitedDate[i];
+        break;
+
+      case 'MM':
+        month = splitedDate[i];
+        break;
+
+      case 'DD':
+        day = splitedDate[i];
+        break;
+    }
+  }
+
+  const result = [];
+
+  const correctYear = transformYear(year, fromFormat, toFormat);
```

Comment:
`transformYear` function is too complicated
you may check which year format you have at the moment and just add another one to your object